tools/toolchains/mbed_toolchain.py

```diff
@@ -395,29 +395,48 @@ def get_arch_file(self, objects):
         cmd_list = (c.replace("\\", "/") for c in objects if c)
         return self.make_option_file(list(cmd_list), ".archive_files.txt")
 
+    def compile_legacy_sources(
+        self, resources, inc_dirs=None, exclude_dirs=None
+    ):
+        """Compile source files with option to exclude some directories.
+
+        This method only exists to not break API compatibility and provide a
+        way to exclude directories for Mbed OS 2 builds.
+        """
+        return self._compile_sources(
+            resources, inc_dirs=inc_dirs, exclude_dirs=exclude_dirs
+        )
+
     # THIS METHOD IS BEING CALLED BY THE MBED ONLINE BUILD SYSTEM
     # ANY CHANGE OF PARAMETERS OR RETURN VALUES WILL BREAK COMPATIBILITY
-    def compile_sources(self, resources, inc_dirs=None, exclude_paths=None):
+    def compile_sources(self, resources, inc_dirs=None):
+        """Compile source files."""
+        return self._compile_sources(resources, inc_dirs=inc_dirs)
+
+    def _exclude_files_from_build(self, files_to_compile, exclude_dirs):
```

Comment:
This should be a static method as it doesn't use self
```suggestion
    @staticmethod
    def _exclude_files_from_build(files_to_compile, exclude_dirs):
```